### PR TITLE
More precise types on requests and reservations

### PR DESCRIPTION
**Pull request by spaceo**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-99

#### Description

Set more precise types on requestsAndReservations
`requests`was not inferred correctly.

### CODE REVIEW

File: src/apps/reservation-list/modal/delete-reservation/helper.ts [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+export type OperationPhysical = UseMutateFunction<
+  void | null,
+  unknown,
+  ParamsPhysical,
+  unknown
+>;
+type Request = {
+  params: { params: { reservationid: number[] } } | { identifier: string };
+  operation: OperationPhysical | OperationDigital;
+};
```
Comment (kasperg) [inline src/apps/reservation-list/modal/delete-reservation/helper.ts:33]:
```suggestion
type Request = {
  params: { params: { reservationid: number[] } } 
  operation: OperationPhysical
} | {
  params: { identifier: string }
  operation: OperationDigital
};
```

If we want to make types more precise consider matching parameter with operation type.

File: src/apps/reservation-list/modal/delete-reservation/helper.ts [hunk 2]
```diff
@@ -22,28 +48,6 @@ export const getReservationsToDelete = (reservations: ReservationType[]) => {
...
-  if (physical === "error" || digital === "error") {
-    return "error";
-  }
-
-  if (physical === "success" || digital === "success") {
-    return "success";
-  }
-
-  return "idle";
-};
```
Comment (kasperg) [inline src/apps/reservation-list/modal/delete-reservation/helper.ts:45]: Nice to get rid of this.

**Review by kasperg — APPROVED** [review]
:+1: I am not sure what the problem is by looking at the code and reading the comment but it seems reasonable.

If we are talking precise types I have one suggestion for you.

File: src/apps/reservation-list/modal/delete-reservation/helper.ts (hunk 1 above)
Comment (spaceo) [reply src/apps/reservation-list/modal/delete-reservation/helper.ts:33]: Brilliant! Of course. Thx 👍

**Pull request merged**